File: go/bin_to_decimal.go
// https://www.codewars.com/kata/57a5c31ce298a7e6b7000334
// 
// attempt #1:
package kata

import (
  "math"
  "strconv"
)

func BinToDec(bin string) int {
  x := len(bin) - 1
  result := 0

  for _, bit := range bin {
    b, _ := strconv.Atoi(string(bit))
    result += b * int(math.Pow(float64(2), float64(x)))
    x -= 1
  }
  return result
}
//
// attempt #2:
package kata

import "strconv"

func BinToDec(bin string) int {
  num, _ := strconv.ParseInt(bin, 2, 64)
  return int(num)
}

File: python/bin_to_decimal.py
# https://www.codewars.com/kata/57a5c31ce298a7e6b7000334
#
# attempt #1:
def bin_to_decimal(inp):
    result = 0
    p = len(inp) - 1
    
    for i in inp:
        result += int(i) * 2 ** p
        p -= 1
    
    return result
#
# attempt #2:
def bin_to_decimal(inp):
    return int(inp, 2)
